### PR TITLE
Possible solution to #31

### PR DESCRIPTION
**Pull request by johnoliver**
possible solution to #31

##### Checklist

- [x] `mvn clean package` build and test completes
- [ ] documentation is changed or added

### CODE REVIEW

File: adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt [hunk 1]
```diff
@@ -2,10 +2,7 @@ package net.adoptopenjdk.api.v3.mapping.upstream
 
 import net.adoptopenjdk.api.v3.dataSources.github.graphql.models.GHRelease
 import net.adoptopenjdk.api.v3.mapping.ReleaseMapper
-import net.adoptopenjdk.api.v3.models.Release
-import net.adoptopenjdk.api.v3.models.ReleaseType
-import net.adoptopenjdk.api.v3.models.Vendor
-import net.adoptopenjdk.api.v3.models.VersionData
+import net.adoptopenjdk.api.v3.models.*
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt]: explicit  imports

File: adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt [hunk 2]
```diff
@@ -46,13 +43,24 @@ object UpstreamReleaseMapper : ReleaseMapper() {
                 versionData = getVersionData(release_name)
             }
 
-            return Release(release.id, release_type, release_link, release_name, timestamp, updatedAt, binaries.toTypedArray(), download_count, vendor, versionData)
+            val sourcePackage = getSourcePackage(release)
+
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt]: remove extra whitespace

File: adoptopenjdk-api-v3-updater/src/test/kotlin/net/adoptopenjdk/api/UpstreamReleaseMapperTest.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import kotlinx.coroutines.runBlocking
+import net.adoptopenjdk.api.v3.dataSources.github.graphql.models.GHAsset
+import net.adoptopenjdk.api.v3.dataSources.github.graphql.models.GHAssets
+import net.adoptopenjdk.api.v3.dataSources.github.graphql.models.GHRelease
+import net.adoptopenjdk.api.v3.dataSources.github.graphql.models.PageInfo
+import net.adoptopenjdk.api.v3.mapping.upstream.UpstreamReleaseMapper
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/test/kotlin/net/adoptopenjdk/api/UpstreamReleaseMapperTest.kt]: remove extra line

File: adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt [hunk 4]
```diff
@@ -46,13 +43,24 @@ object UpstreamReleaseMapper : ReleaseMapper() {
...
+            return Release(release.id, release_type, release_link, release_name, timestamp, updatedAt, binaries.toTypedArray(), download_count, vendor, versionData, sourcePackage)
         } catch (e: FailedToParse) {
             LOGGER.error("Failed to parse $release_name")
             return null
         }
     }
 
+    private fun getSourcePackage(release: GHRelease): SourcePackage? {
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt]: We may need to widen this definition later, but let's see.

**Comment by johnoliver** [comment]
@karianna should be good

**Comment by johnoliver** [comment]
@karianna can you hold off on merging this for a sec

**Pull request merged**